### PR TITLE
[14.0][IMP] sale_order_revision: 'New Revision' quick button on confirmed Sales Orders

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

**Review by rousseldenis — COMMENTED** [review]
LGTM. @dreispt Could you add a little test ?

**Comment by dreispt** [comment]
@rousseldenis Test added!

File: sale_order_revision/models/sale_order.py [hunk 1]
```diff
@@ -42,3 +42,10 @@ def action_view_revisions(self):
             "default_current_revision_id": self.id,
         }
         return result
+
+    def action_cancel_create_revision(self):
+        for sale in self:
+            sale.action_cancel()
+            action = sale.create_revision()
+        if len(self) == 1:
```
Comment (rousseldenis) [inline sale_order_revision/models/sale_order.py:50]: @dreispt You should always return something in action buttons. This prevent to be overridden

Comment (dreispt) [reply sale_order_revision/models/sale_order.py:50]: Done!

**Review by rousseldenis — APPROVED** [review]

File: sale_order_revision/view/sale_order.xml [hunk 2]
```diff
@@ -23,6 +23,12 @@
                     string="New Revision of Quotation"
                     type="object"
                 />
+                <button
+                    name="action_cancel_create_revision"
+                    states="sale"
+                    string="New Revision"
```
Comment (rousseldenis) [inline sale_order_revision/view/sale_order.xml:29]: I would have added something to help user to know the current one will be canceled (in string or in help at least)

**Review by rousseldenis — COMMENTED** [review]
@dreispt Could you also improve README ?

**Comment by dreispt** [comment]
README improved.

**Comment by dreispt** [comment]
@rousseldenis Can this be merged?

**Comment by dreispt** [comment]
Ping @rousseldenis

**Comment by dreispt** [comment]
Based on [this](https://github.com/OCA/sale-workflow/pull/2930#pullrequestreview-1852058329) comment, reworked and superseded by #2930

**Pull request closed without merging**